designdocs/batched_messages.md

```diff
@@ -10,92 +10,148 @@
+and payload location.
 
-This document does not describe the types of DSNP messages that would be in batch announcements. Batch announcements must reference a schema through its ID number. A schema may or may not describe a DSNP message.
+This document also specifies how schemas will constrain the shape of off chain
+messages.
 
-This document does not discuss validation of batch file format located at the URI in the announcement, since
-the file format cannot be verified on-chain. For details about batch files themselves, see [DSNP Spec: Batch Publications](https://spec.dsnp.org/DSNP/BatchPublications).
+This document does not describe the types of DSNP messages that will be
+described by schemas. In theory, any message model can be supported.
+
+This document also does not discuss validation of either model or model type. If
+this type of validation is necessary, it should be described elsewhere.
 
 ## Proposal
 * All names are placeholders and may be changed.
 * Types may change as needed during implementation phase
 * Errors in the extrinsic(s) must have different, reasonably-named error enums for each type of error for ease of debugging.
 
 ### Constants
-* `BatchPageSizeMax` - the maximum number of batch announcements that will be returned by the `get_batches` query
-* `BatchSizeMinBytes` - the minimum possible size of a valid batch file with 1 row
+TBD
 
 ### Enums
-* `BatchFormat` - a list of supported formats for batch files. Currently only Parquet file format is supported, however,
-other formats are being considered.
-
-### Types
-* `BatchAnnouncementParams<T:Config>`: generic
-    * `batch_uri`:`Vec<u8>` the URI of the batch file. Must be an IPFS [CIDv1](https://github.com/multiformats/cid/) URI. Accepted codec, algorithm, and base are to be determined.
-    * `message_schema_id`: `SchemaId`  the schema id for the messages in this batch. The `schema_id` must refer to schemas used for batching only.
-    * `file_size`: `usize`, the size of the batch file, used to determine message fee as well as to let consumers know what size files to expect.  Must be &gt;= the minimum possible DSNP batch file size.
-    * `file_format`: `BatchFormat`, indicator of the file format of the batch file.
-
-The file hash is not included separately. Since the `batch_uri` uses CIDv1 specification, the file hash is already included.
+* `ModelType` - supported serialization formats for message payloads files. Currently only [Parquet](https://parquet.apache.org/docs/) and
+  [Avro](https://avro.apache.org/docs/current/) are supported.
+* `PayloadLocation` - The location of the payload. Can be either `OnChain` or `IPFS`.
 
-* `BatchAnnouncement`: implements `BatchAnnouncementParams`, returned by `get_batches`
+### Traits
+* `Model` - TBD. A set of functions for accessing message payload information.
+  * Derives `Encode`, `Decode`, `MaxEncodedLen`
+  * `max_length`: `SchemaMaxBytesBoundedVecLimit`
```

Comment:
I'm not sure what this trait is for.

Reply:
Ah I see as `max_length` might be different for different Model Types right?

Reply:
yes, that and payload